addons/web_editor/static/lib/odoo-editor/src/commands/deleteBackward.js

```diff
@@ -202,7 +205,7 @@ HTMLElement.prototype.oDeleteBackward = function (offset, alreadyMoved = false,
             currentNodeIndex++;
         }
     } else {
-        currentNodeIndex = offsetLimit;
+        currentNodeIndex = Math.min(offsetLimit, this.childNodes.length);
```

Comment:
My apologies, I realized that I could reproduce the second traceback only in 16.0. It may explain why you were not able to do it.

Reply:
Sorry for the late reply, this PR seem to have slipped under the radar in my inbox for some reason. I could indeed reproduce the issue on a 16.0 runbot.

I don't think think the offsets given as parameter to the function should be corrected because then the function is going to do something different than what it was asked to do. It is arguable whether that is a big deal or not since what the function was asked to do does not make sense since the offset is wrong. However, I personally would except the function to traceback in such a case, not somehow correct the values I gave them and do _something else_ than what I asked. Anyway, my point is that I think we need to find where the wrong offset originates from. I think I might have found at least one of the causes.

In `moveNodes`, the function might return the position it received as parameters, but this position might not be valid anymore after the move. In particular, the position is invalid if `sourceEl` is the only child of `destinationEl` and that no nodes were actually moved, which is precisely what happens in your test case. Therefore, I suggest replacing the return in `moveNodes` by the following:
```js
return firstNode
    ? leftPos(firstNode)
    : [
        destinationEl,
        Math.min(destinationOffset, destinationEl.childNodes.length),
    ];
```

I don't get the traceback anymore with this change but since it is not exactly clear to me what cases you have seen in practice that trigger this bug, I cannot be sure the end result matches what you expected. Could you give it a try and let me know please ?

Reply:
I think the problem is that `nodes` is `[undefined]`, which tricks the `nodes.length` condition to be true even though there aren't actually any node. I think giving a value of 2 to `endIndex` does not make any sense. The value 1 has a meaning because it means "after" the only child node, but 2 is just an invalid position IMO. Therefore, I suspect the next mistake to fix is in the code that computes this value of 2 in that particular case.

Reply:
I came back to this issue recently. I found that it happens only when I copy & paste content from Google Docs. When I copy the same thing from Microsoft Word the error does not appear.

On the following screenshot, you can see that for the same content, it produces completely different HTML output. The HTML from Google Docs in particular feels kinda odd given that it puts `<p>` inside another `<p>`. I am not very familiar with the code from the odoo editor so I am not sure if this can be the root of the error but that might be a good starting point. Let me know what you think.
![image](https://github.com/odoo/odoo/assets/46035353/6141c3ad-921b-48b1-ab58-7dbc9d13dc21)

Also while investigating, I saw that when pasting multiple paragraphs from Google Docs, then CTRL + A, then BackSpace, it goes stuck [in this loop](https://github.com/odoo/odoo/blob/9ffe68e590e97f12bac067abffb22b512128344f/addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js#L1947) and the page freezes. So, there is obviously something wrong while formatting the text pasted from G Docs.